.swiftlint.yml

```diff
@@ -76,3 +76,10 @@ custom_rules:
     regex: 'NSLocalizedString\("[^"]*\\\(\S*\)'
     message: "Localized strings must not use interpolated variables. Instead, use `String(format:`"
     severity: error
+
+  swiftui_localization:
+    name: "SwiftUI Localization"
+    regex: 'LocalizedStringKey'
+    message: "Using `LocalizedStringKey` is incompatible with our tooling and doesn't allow you to provide a hint/context comment for translators either. Please use `NSLocalizedString` instead, even with SwiftUI code."
+    severity: error
+    excluded: '.*Widgets/.*'
```

Comment:
Widgets are excluded from that rule because we haven't decided how we handle their case yet (ref: paaHJt-sd-p2#comment-3600).

See also [past discussion on this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/15614#pullrequestreview-565691006) between @Gio2018 , @jkmassel and myself.